src/models/meta_details.rs

```diff
@@ -189,6 +201,55 @@ fn library_item_sync(library_item: &Option<LibraryItem>, profile: &Profile) -> E
     }
 }
 
+fn selected_override_update(
+    selected: &mut Option<Selected>,
+    meta_items: &Vec<ResourceLoadable<MetaItem>>,
+) -> Effects {
+    let meta_path = if let Some(Selected {
+        meta_path,
+        stream_path: None,
+    }) = &selected
+    {
+        meta_path
+    } else {
+        return Effects::default();
+    };
+    let meta_item = if let Some(meta_item) = meta_items
+        .iter()
+        .find_map(|meta_item| match &meta_item.content {
+            Some(Loadable::Ready(meta_item)) => Some(Some(meta_item)),
+            Some(Loadable::Loading) => Some(None),
+            _ => None,
+        })
+        .flatten()
```

Comment:
can be simplified and remove the `.flatten()` by just removing the `Some(Some(..))` & `Some(None)`